src/gui/src/SampleEditor/TargetWaveDisplay.cpp

```diff
@@ -204,45 +206,54 @@ void TargetWaveDisplay::paintLocatorEventTargetDisplay( int pos, bool updateposi
 
 void TargetWaveDisplay::updateDisplay( std::shared_ptr<H2Core::InstrumentLayer> pLayer )
 {
-	if ( pLayer && pLayer->get_sample() ) {
-
-		int nSampleLength = pLayer->get_sample()->get_frames();
-		float nScaleFactor = nSampleLength / width();
-
-		float fGain = (height() - 8) / 2.0 * pLayer->get_gain();
-
-		auto pSampleDatal = pLayer->get_sample()->get_data_l();
-		auto pSampleDatar = pLayer->get_sample()->get_data_r();
-		int nSamplePos = 0;
-		int nVall;
-		int nValr;
-		for ( int i = 0; i < width(); ++i ){
-			nVall = 0;
-			nValr = 0;
-			for ( int j = 0; j < nScaleFactor; ++j ) {
-				if ( j < nSampleLength ) {
-					if ( pSampleDatal[ nSamplePos ] < 0 ){
-						int newVal = static_cast<int>( pSampleDatal[ nSamplePos ] * -fGain );
-						nVall = newVal;
-					}else
-					{
-						int newVal = static_cast<int>( pSampleDatal[ nSamplePos ] * fGain );
-						nVall = newVal;
-					}
-					if ( pSampleDatar[ nSamplePos ] > 0 ){
-						int newVal = static_cast<int>( pSampleDatar[ nSamplePos ] * -fGain );
-						nValr = newVal;
-					}else
-					{
-						int newVal = static_cast<int>( pSampleDatar[ nSamplePos ] * fGain );
-						nValr = newVal;
-					}
+	// qWarning() << "TargetWaveDisplay::updateDisplay: layer:" << pLayer;
+	if ( ! pLayer ) {
+		return;
+	}
+	updateDisplay( pLayer->get_sample(), pLayer->get_gain() );
+}
+
+void TargetWaveDisplay::updateDisplay( std::shared_ptr<H2Core::Sample> sample, double gain )
+{
+	// qWarning() << "TargetWaveDisplay::updateDisplay: sample:" << sample;
```

Comment:
The original version of `updateDisplay` also included a check whether `pLayer->get_sample()` is a `nullptr`. Could you add it in here too just to be save?